backend/franky/service/dockerhub/dockerhub.py

```diff
@@ -0,0 +1,172 @@
+from urllib.parse import urljoin
+from typing import Optional, List, Dict
+from datetime import datetime, timedelta
+
+import requests
+
+from model.user import UserData
+from model.project import ProjectData
+from service import Service
+
+
+class DockerHubException(Exception):
+    pass
+
+
+def user_cleaner(username: str) -> str:
+    """
+    Makes username lowercase
+
+    :param username: user login
+    :return: cleaned username
+    """
+    return username.lower()
+
+
+class DockerHub(Service):
+
+    def __init__(self, rest_endpoint='https://hub.docker.com'):
+        self._version = 'v2'
+        self._rest_endpoint = rest_endpoint
+        self._url = '%s/%s' % (self._rest_endpoint, self._version)
+        self._user_url = 'users'
+        self._tag_url = 'tags'
+        self._repository_url = 'repositories'
+        self._tag_indexes = ['os', 'architecture']
+        self._date_format = '%Y-%m-%dT%H:%M:%SZ'
+        self._active_projects_period = timedelta(weeks=2)
+        self._page_size = 25
+
+    def user(self, username: str) -> UserData:
+        login = user_cleaner(username)
+        full_name = self._get_user_full_name(username)
+        tags = self._get_user_tags(username)
+        return UserData(login=login, name=full_name, tags=tags)
+
+    def _get_user_full_name(self, username) -> str:
+        username = user_cleaner(username)
+        request_url = self._api_url('%s/%s' % (self._user_url, username))
+        payload = dict()
+        response_data = self._call(request_url, payload)
+        return response_data['full_name']
+
+    def _get_user_tags(self, username: str) -> Dict:
+        username = user_cleaner(username)
+        request_url = self._api_url('%s/%s' % (self._repository_url, username))
+        payload = {'page_size': self._page_size, 'page': 1, 'ordering': 'last_updated'}
+        all_tags = {}
+        while True:
+            response_data = self._call(request_url, payload)
+            repositories = response_data['results']
+            for repository in repositories:
+                image_name = repository['name']
+                tags = self._get_tags(username, image_name)
+                if tags:
+                    for tag, byte_of_code in tags.items():
+                        all_tags[tag] = all_tags.get(tag, 0) + byte_of_code
+            if not response_data['next']:
+                break
+            else:
+                payload['page'] += 1
+        total_bytes = sum(all_tags.values())
+        return {tag: byte_of_code / total_bytes for tag, byte_of_code in all_tags.items()}
+
+    def _get_tags(self, username, image_name: str) -> Optional[Dict]:
```

Comment:
The method cannot return `None` therefore its returning type can be specified as `Dict` rather than `Optional[Dict]`.